src/lib/socket.c

```diff
@@ -57,6 +57,42 @@
 #include <openssl/ssl.h>
 #include <openssl/err.h>
 #include <openssl/bio.h>
+#if OPENSSL_VERSION_NUMBER < 0x10100001L
+#undef BIO_ctrl_set_connected
+#define BIO_ctrl_set_connected(b, peer) (int)BIO_ctrl(b, BIO_CTRL_DGRAM_SET_CONNECTED, 0, (char *)peer)
```

Comment:
Why remove the 2nd parameter? Evening ignoring OpenSSL 0.9.8, OpenSSL 1.0 won't be going away anytime soon.

Did you try compiling against OpenSSL 1.0? I'm surprised the code at 1445:1481 compiles with 1.0.